security/l1-contracts-dev-flow-vision.md

```diff
@@ -0,0 +1,159 @@
+source code will require no changes to this method—for example, if you only change things like
+code comments or function internals. Other changes will require changes to this method, like
+adding new contracts or modifying the `initialize` signature. For the latter case, because OPCM
+is what's used to scaffold chains for testing, you will be forced to modify the `deploy` method to
+account for your changes in order to test them. This is great, as it's now trivial to keep deploy
+scripts up to date, and we essentially get this for free.
+
+OPCM will also contain an `upgrade` method as described in the [OPCM Upgrades design doc](l1-upgrades.md).
+Some basic unit tests will test this method. Additionally, a flag can be set to have tests
+run against the state resulting from running `OPCM.upgrade` against an existing chain (instead of the
+default where tests are run against the state resulting from a fresh deploy via `OPCM.deploy()`). CI will be
+responsible for running the tests against the `upgrade` method for all mainnet and sepolia chains
+that the Security Council + Optimism Foundation (the 2/2) hold keys for. This enforces that
+developers are also writing the upgrade logic at the same time, to provide guarantees that the full
+test suite passes for new deploys and upgrades for all existing chains.
+
+## Releases and Contract Deployments
+
+After development, we are now ready to cut a release candidate and deploy the new implementation
+contracts to a chain, such as devnet, sepolia, or mainnet. All contract deployments must be done
+against a tag of the form `op-contracts/vX.Y.Z[-rc.n]`. To deploy the contract, create a `bootstrap`
+command in op-deployer, and run that new command to deploy the new contracts.
+
+An alternate approach that we might support that avoids the need for bootstrap commands is using
+CREATE2 in the `DeploySuperchain.s.sol` and `DeployImplementations.s.sol` scripts. This way,
+deploying contracts is as easy as re-running those scripts—if bytecode for a contract has not changed,
+the resulting create2 address is the same, so we can skip deploying that unchanged contract.
+
+The current [release and versioning process](https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/meta/VERSIONING.md)
+is unintuitive and requires a lot of manual work. It is not possible to have the source code for the
+entirety of a release at a single commit, and therefore very difficult to test contract releases.
+Additionally, deploy and upgrade scripts are not part of a release.
+
+Now, OPCM serves as the source of truth for what a release is. A release is defined as the OPCM contract
+itself, along with all contracts that are used in the `deploy()` or `upgrade()` methods of OPCM.
+
+## Execution of Onchain Transactions
+
+With new implementation contracts deployed, we can prepare playbooks, or tasks, in superchain-ops.
```

Comment:
Can we move away from gnosis safe JSON being the input to the superchain-ops tasks? We don't use the UI anyways, its not safe enough to use because it doesn't allow for the various security checks that we do (local simulation + assertions, observe the various hashes and compare to ledger).

Looking at the work that the protocol team has to do to generate the JSON for the superchain-ops task (https://github.com/ethereum-optimism/optimism/pull/13412) it is WAY to complex. We need something that is dead simple. It sounds like we will just write a forge script now? And have access to a "superchain-ops standard library" of sorts?

Reply:
Yes, writing forge scripts is the plan. From disussions with @mds1, we want to get as close as possible to having everything just be a template. 
Even upgrades should _mostly_ reuse code from an upgrade library, although it will vary somewhat depending on what new config values are introduced.